pythonforandroid/recipes/python3/__init__.py

```diff
@@ -33,13 +33,53 @@
 
 
 class Python3Recipe(TargetPythonRecipe):
+    '''
+    .. note::
+        In order to build certain python modules, we need to add some extra
+        recipes to our build requirements:
+
+            - ctypes: you must add the recipe for ``libffi``.
+    '''
     version = '3.7.1'
     url = 'https://www.python.org/ftp/python/{version}/Python-{version}.tgz'
     name = 'python3'
 
     depends = ['hostpython3']
     conflicts = ['python3crystax', 'python2']
-    # opt_depends = ['openssl', 'sqlite3']
+    opt_depends = ['libffi']  # 'openssl', 'sqlite3'
+
+    configure_args = (
+        '--host={android_host}',
+        '--build={android_build}',
+        '--enable-shared',
+        '--disable-ipv6',
+        'ac_cv_file__dev_ptmx=yes',
+        'ac_cv_file__dev_ptc=no',
+        '--without-ensurepip',
+        'ac_cv_little_endian_double=yes',
+        '--prefix={prefix}',
+        '--exec-prefix={exec_prefix}')
+
+    def set_libs_flags(self, env, arch):
+        '''Takes care to properly link libraries with python depending on our
+        requirements and the attribute :attr:`opt_depends`.
+        '''
+        def add_flags(include_flags, link_flags):
```

Comment:
Since you only use this once it doesn't really need to be a function.

Reply:
You are right, I will fix it